### ISSUE TITLE
Mismatched type between "Result" and "i32" is not reported

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#4595]
## Environment

* **IntelliJ Rust plugin version:** 0.2.110.2779-193-nightly
* **Rust toolchain version:** 1.40.0-nightly (87cbf0a54 2019-11-01) x86_64-apple-darwin
* **IDE name and version:** CLion 2020.1 Snapshot (CL-201.706)
* **Operating system:** macOS 10.15

## Problem description

Plugin doesn't report mismatched type for `Result` when `i32` is expected.

## Steps to reproduce

E.g.

```rust
pub fn string_to_number(s: &str) -> i32 {
    s.parse()
}
```

The plugin reports no problem while the code will fail to compile. Not exactly sure why there is no error here.
